File: _includes/code/csharp/quickstart/GitHubReadmeExample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json;
using System.Threading.Tasks;
using Weaviate.Client;
using Weaviate.Client.Models;

namespace WeaviateProject.Examples
{
    public class GitHubReadmeExample
    {
        public static async Task Run()
        {
            // Connect to Weaviate
            // Using try-with-resources ensures client.close() is called automatically
            var client = await Connect.Local();

            // Clean slate (not in original script, but helpful for re-running main methods)
            if (await client.Collections.Exists("Article"))
            {
                await client.Collections.Delete("Article");
            }

            // Create a collection
            var articles = await client.Collections.Create(
                new CollectionCreateParams
                {
                    Name = "Article",
                    Properties = [Property.Text("content")],
                    VectorConfig = Configure.Vector("default", v => v.Text2VecTransformers()), // Use a vectorizer to generate embeddings during import
                    // VectorConfig = Configure.Vector("default", v => v.SelfProvided())  // If you want to import your own pre-generated embeddings
                }
            );

            // Insert objects and generate embeddings
            var data = new List<object>
            {
                new { content = "Vector databases enable semantic search" },
                new { content = "Machine learning models generate embeddings" },
                new { content = "Weaviate supports hybrid search capabilities" },
            };
            await articles.Data.InsertMany(data.ToArray());

            await Task.Delay(1000);
            // Perform semantic search
            var results = await articles.Query.NearText("Search objects by meaning", limit: 1);
            // Print result
            if (results.Objects.Count > 0)
            {
                Console.WriteLine(JsonSerializer.Serialize(results.Objects.First()));
            }
        }
    }
}

File: _includes/code/java-v6/src/test/java/quickstart/GitHubReadmeExample.java
package quickstart;

import io.weaviate.client6.v1.api.WeaviateClient;
import io.weaviate.client6.v1.api.collections.CollectionHandle;
import io.weaviate.client6.v1.api.collections.Property;
import io.weaviate.client6.v1.api.collections.VectorConfig;
import java.util.Map;

public class GitHubReadmeExample {

  public static void main(String[] args) throws Exception {
    // Connect to Weaviate
    // Using try-with-resources ensures client.close() is called automatically
    try (WeaviateClient client = WeaviateClient.connectToLocal()) {

      // Clean slate (not in original script, but helpful for re-running main methods)
      if (client.collections.exists("Article")) {
        client.collections.delete("Article");
      }

      // Create a collection
      client.collections.create("Article",
          col -> col.properties(Property.text("content"))
              .vectorConfig(VectorConfig.text2vecTransformers()) // Use a vectorizer to generate embeddings during import
           // .vectorConfig(VectorConfig.selfProvided()) // If you want to import your own pre-generated embeddings
      );

      // Insert objects and generate embeddings
      CollectionHandle<Map<String, Object>> articles =
          client.collections.use("Article");

      articles.data.insertMany(
          Map.of("content", "Vector databases enable semantic search"),
          Map.of("content", "Machine learning models generate embeddings"),
          Map.of("content", "Weaviate supports hybrid search capabilities"));

      Thread.sleep(1000);
      // Perform semantic search
      var results =
          articles.query.nearText("Search objects by meaning", q -> q.limit(1));

      // Print result
      if (!results.objects().isEmpty()) {
        System.out.println(results.objects().get(0));
      }
    }
  }
}
